src/source.cpp

```diff
@@ -47,9 +47,20 @@ vector<unique_ptr<Source>> external_sources;
 }
 
 //==============================================================================
-// Source create implementation
+// Source implementation
 //==============================================================================
 
+Source::Source(pugi::xml_node node)
+{
+  // Check for source strength
+  if (check_for_node(node, "strength")) {
+    strength_ = std::stod(get_node_value(node, "strength"));
```

Comment:
Should we check that this is > 0? Negative should always be forbidden. Zero would deserve a warning, maybe.